File: server/src/test/java/com/fluffy/reaction/domain/ReactionTest.java
package com.fluffy.reaction.domain;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatCode;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class ReactionTest {

    @Test
    @DisplayName("반응을 정상적으로 생성할 수 있다.")
    void create() {
        // when & then
        assertThatCode(() -> new Reaction("EXAM", 1L, 1L, ReactionType.LIKE))
                .doesNotThrowAnyException();
    }

    @Test
    @DisplayName("반응을 삭제 상태로 변경할 수 있다.")
    void delete() {
        // given
        Reaction reaction = new Reaction("EXAM", 1L, 1L, ReactionType.LIKE);

        // when
        reaction.delete();

        // then
        assertThat(reaction.getStatus()).isEqualTo(ReactionStatus.DELETED);
    }

    @Test
    @DisplayName("삭제 상태의 반응을 활성 상태로 변경할 수 있다.")
    void active() {
        // given
        Reaction reaction = new Reaction("EXAM", 1L, 1L, ReactionType.LIKE);
        reaction.delete();

        // when
        reaction.active();

        // then
        assertThat(reaction.getStatus()).isEqualTo(ReactionStatus.ACTIVE);
    }
}

File: server/src/test/kotlin/com/fluffy/reaction/domain/ReactionTest.kt
package com.fluffy.reaction.domain

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class ReactionTest : StringSpec({

    "반응은 정상적으로 생성된다." {
        // given
        val reaction = Reaction.create(
            targetType = "EXAM",
            targetId = 1L,
            memberId = 1L,
            type = ReactionType.LIKE,
        )

        // when & then
        reaction.status shouldBe ReactionStatus.ACTIVE
    }

    "반응을 삭제하면 삭제 상태가 된다." {
        // given
        val reaction = Reaction.create(
            targetType = "EXAM",
            targetId = 1L,
            memberId = 1L,
            type = ReactionType.LIKE,
        )

        // when
        reaction.delete()

        // then
        reaction.status shouldBe ReactionStatus.DELETED
    }

    "반응을 활성화하면 활성 상태가 된다." {
        // given
        val reaction = Reaction.create(
            targetType = "EXAM",
            targetId = 1L,
            memberId = 1L,
            type = ReactionType.LIKE,
        )
        reaction.delete()

        // when
        reaction.activate()

        // then
        reaction.status shouldBe ReactionStatus.ACTIVE
    }
})